nav2_behavior_tree/plugins/decorator/is_new_path_shorter.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2022 Pradheep Padmanabhan - Neobotix GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <vector>
+#include "nav2_util/geometry_utils.hpp"
+
+#include "nav2_behavior_tree/plugins/decorator/is_new_path_shorter.hpp"
+
+namespace nav2_behavior_tree
+{
+
+IsNewPathShorter::IsNewPathShorter(
```

Comment:
I think this implementation is also flawed, you tick the children with the path is **longer** not shorter... when shorter you just return success. I don't think this implementation semantically makes sense by its name / BT type as a decorator

Reply:
It's the name that is causing the confusion. I will try to come up with a new name that semantically makes sense for the BT.